src/screens/login/Login.tsx

```diff
@@ -1,115 +1,44 @@
-  const isDark = useAppSelector(({ theme }) => theme.isDark);
-  const { saveAccessToken, saveRefreshToken, login } = useContext(TokenContext);
-
-  const [loading, setLoading] = useState(false);
-  const [loadingProcessInfo, setLoadingProcessInfo] = useState('');
+  const { loading, loadingProcess, error } = useAppSelector(({ user }) => user);
+  const dispatch = useAppDispatch();
 
   const globalStyles = GlobalThemedStyles();
   const styles = ThemedStyles();
 
-  const {
-    CLIENT_ID,
-    ACCOUNT_BASE_URL,
-    data,
-    endpoints: { account: accountEndpoints },
-  } = ApiConstants;
-
-  const getUserAuthorization = async () => {
-    setLoading(true);
-    setLoadingProcessInfo('Authorizing User');
-    const { redirectUrl, authorizationScope, authResponseType } = data.account;
-    const { requestAuthorization: requestAuthorizationEndpoint } = accountEndpoints;
-
-    const state = generateRandomString(16);
-
-    const searchParams: UserAuthorizationParams = {
-      client_id: CLIENT_ID,
-      redirect_uri: redirectUrl,
-      response_type: authResponseType,
-      scope: authorizationScope,
-      state,
-    };
-
-    const url = new URL(requestAuthorizationEndpoint, ACCOUNT_BASE_URL);
-    appendSearchParams<UserAuthorizationParams>(url, searchParams);
-
-    const urlString = url.toString();
-    const isUrlValid = await Linking.canOpenURL(urlString);
-
-    if (!isIos || isUrlValid) {
-      await Linking.openURL(urlString);
-    } else {
-      setLoading(false);
-      Toast.show({ text1: 'Invalid Auth Url', type: 'error' });
-    }
-  };
-
-  const getAccessToken = async (code: string) => {
-    setLoadingProcessInfo('Fetching Token');
-    const { requestAccessToken: requestAccessTokenEndpoint } = accountEndpoints;
-
+  const handleDeepLink = async ({ url }: { url: string }) => {
     const {
-      grantType: { code: codeGrantType },
-      redirectUrl,
-    } = data.account;
-
-    const body: RequestAccessTokenBody = {
-      grant_type: codeGrantType,
-      redirect_uri: redirectUrl,
-      code,
-    };
-
-    const response = await _postAccount<AuthAccessTokenResponse, RequestAccessTokenBody>(
-      requestAccessTokenEndpoint,
-      body,
-    );
-
-    if (!response.success) {
-      setLoading(false);
-      const { message } = response.error;
-      Toast.show({ text1: `Error ${code}`, text2: message, type: 'error' });
-      return;
-    }
+      searchParams: { code },
+    } = parseUrl<AuthCodeResponseUrlType>(url);
 
-    const { access_token, expires_in, refresh_token, token_type } = response.responseData;
-    saveAccessToken(`${token_type} ${access_token}`, expires_in);
-    saveRefreshToken(refresh_token);
+    const { meta, payload } = await dispatch(requestAccessTokenViaCode(code));
 
-    setLoading(false);
-    login();
-  };
+    if (meta.requestStatus === REQUEST_STATUS.FULFILLED) {
+      const { access_token, refresh_token, token_type } = payload as AuthAccessTokenResponse;
```

Comment:
why `payload` here if this a response, its confusing with payload

Reply:
same as above

Reply:
i have created separate thunk for it